Sources/GRPC/ClientConnection.swift

```diff
@@ -426,16 +431,31 @@ extension Channel {
     #endif
 
     if let tlsConfiguration = tlsConfiguration {
-      do {
-        let sslClientHandler = try NIOSSLClientHandler(
-          context: try NIOSSLContext(configuration: tlsConfiguration),
-          serverHostname: tlsServerHostname
-        )
-        handlers.append(sslClientHandler)
-        handlers.append(TLSVerificationHandler(logger: logger))
-      } catch {
-        return self.eventLoop.makeFailedFuture(error)
-      }
+        if let customVerificationCallback = customVerificationCallback {
```

Comment:
There's a lot of duplicated code here: let's move the if-else inside the do-catch so that we don't duplicate the error handling or adding of handlers.